header/p2p/exchange.go

```diff
@@ -154,9 +206,41 @@ func (ex *Exchange) performRequest(
 
 		headers[i] = header
 	}
+	if err = stream.Close(); err != nil {
+		log.Errorw("closing stream", "err", err)
+	}
 	// ensure at least one header was retrieved
 	if len(headers) == 0 {
 		return nil, header.ErrNotFound
 	}
-	return headers, stream.Close()
+	return headers, nil
+}
+
+// bestHead chooses ExtendedHeader that matches the conditions:
+// * should have max height among received;
+// * should be received at least from 2 peers;
+// If neither condition is met, then latest ExtendedHeader will be returned (header of the highest height).
+func bestHead(result []*header.ExtendedHeader) (*header.ExtendedHeader, error) {
+	if len(result) == 0 {
+		return nil, header.ErrNotFound
+	}
+	counter := make(map[string]int)
+	// go through all of ExtendedHeaders and count the number of headers with a specific hash
+	for i := 0; i < len(result); i++ {
+		counter[result[i].Hash().String()]++
+	}
```

Comment:
q: why nor for range?

Reply:
No reasons... Will make it more go idiomatic